### ISSUE TITLE
feat: Validation mechanism to limit "free traffic" on the network

### ISSUE THREAD

**Issue by chair28980** [waku-org/nwaku#1938]
This tracks the work to define and implement a relay validator that allows nodes to limit the amount of bandwidth they're willing to allocate for routing "free traffic". In this case, "free traffic" would be Waku messages with no RLN proofs.

Priority: Critical for launch

**Comment by rymnc** [comment]
note: this feature exists within the WakuRlnRelay validator, and needs to be separated from it

before we do that though, we need to ensure ordered execution of validators, which currently happens in parallel in nim-libp2p

1. WakuMessage decoding
2. Bandwidth rate limiting
3. RLN validator
4. any other validator

**Comment by alrevuelta** [comment]
A clarification on

> they're willing

Nodes can't chose this parameter. It will be hardcoded, to ensure that every node views the same message as valid/invalid. Note still that even with this some messages may be wrongly labeled, due to some minor delays in the network. So not a feature that I really like tbh.

> before we do that though, we need to ensure ordered execution of validators, which currently happens in parallel in nim-libp2p

Im checking this with libp2p team. Quickest solution would be to merge all validators into one so that we can control the order, but checking with them the rationale behind i) running them in paralel and ii) not guaranteeing the order.

**Comment by chair28980** [comment]
Work to be descoped from Gen 0 milestone, per discussion in nwaku pm meeting 2023-11-21.

**Comment by chair28980** [comment]
De-scoped indefinitely until need is established.
